openl3/core.py

```diff
@@ -209,15 +214,327 @@ def process_file(filepath, output_dir=None, suffix=None, model=None,
+                       audio_embedding_size=6144, audio_center=True,
+                       audio_hop_size=0.1, image_embedding_size=8192,
+                       verbose=True):
+    """
+    Computes and saves L3 audio and video frame embeddings for given video file
+
+    Parameters
+    ----------
+    filepath : str
+        Path to video file to be processed.
+    output_dir : str or None
+        Path to directory for saving output files. If None, output files will
+        be saved to the directory containing the input file.
+    suffix : str or None
+        String to be appended to the output filename,
+        i.e. <base filename>_<modality>_<suffix>.npz.
+        If None, then no suffix will be added,
+        i.e. <base filename>_<modality>.npz.
+    audio_model : keras.models.Model or None
+        Loaded audio model object. If a model is provided, then `input_repr`,
+        `content_type`, and `embedding_size` will be ignored.
+        If None is provided, the model will be loaded using
+        the provided values of `input_repr`, `content_type` and
+        `embedding_size`.
+    image_model : keras.models.Model or None
+        Loaded audio model object. If a model is provided, then `input_repr`,
+        `content_type`, and `embedding_size` will be ignored.
+        If None is provided, the model will be loaded using
+        the provided values of `input_repr`, `content_type` and
+        `embedding_size`.
+    input_repr : "linear", "mel128", or "mel256"
+        Spectrogram representation used for audio model. Ignored if `model` is
+        a valid Keras model.
+    content_type : "music" or "env"
+        Type of content used to train embedding. Ignored if `model` is
+        a valid Keras model.
+    audio_embedding_size : 6144 or 512
+        Audio embedding dimensionality. Ignored if `model` is a valid Keras model.
+    audio_center : boolean
+        If True, pads beginning of audio signal so timestamps correspond
+        to center of window.
+    audio_hop_size : float
+        Hop size in seconds.
+    image_embedding_size : 8192 or 512
+        Video frame embedding dimensionality. Ignored if `model` is a valid Keras model.
+    verbose : 0 or 1
+        Keras verbosity.
+
+    Returns
+    -------
+
+    """
+    if not os.path.exists(filepath):
+        raise OpenL3Error('File "{}" could not be found.'.format(filepath))
+
+    try:
+        clip = VideoFileClip(filepath, target_resolution=(256, 256),
+                             audio_fps=TARGET_SR)
+        audio = clip.audio.to_soundarray(fps=TARGET_SR)
+        images = np.array([frame for frame in clip.iter_frames()])
+    except Exception:
+        raise OpenL3Error('Could not open file "{}":\n{}'.format(filepath, traceback.format_exc()))
+
+    if not suffix:
+        audio_suffix = "audio"
+        image_suffix = "image"
+    else:
+        audio_suffix = "audio_" + suffix
+        image_suffix = "image_" + suffix
```

Comment:
not important, but this could be simplified to just:
```
audio_suffix, image_suffix = "audio", "image"
if suffix:
    audio_suffix += "_" + suffix
    image_suffix += "_" + suffix
```

Reply:
Sure.